feat: UIG-2974 - native vl-select - introductie component

ksvlaanderen opened the pull request:
native vl-select-next component toegevoegd
uitgebreide vl-select hernoemd naar vl-select-rich
form-demo uitgebreid
Storybook en Cypress

[Jira](https://www.milieuinfo.be/jira/browse/UIG-2974)
[Bamboo](https://www.milieuinfo.be/bamboo/browse/UIGOV-CUWC327)

Goldflow commented on apps/storybook/docs/e_ontwerp/form/2_demo/form-demo.stories-doc.mdx:
```diff
@@ -14,6 +14,7 @@ De submitted waarden worden in deze demo in de console gelogd.
 - [vl-input-field-masked-next](/docs/form-next-input-field-masked--documentatie)
 - [vl-textarea-next](/docs/form-next-textarea--documentatie)
```
Zouden we hier ook nog link naar textarea-rich-next toevoegen?

Goldflow replied:
Zoals besproken, moet niet.

Goldflow commented on libs/form/src/next/select/stories/vl-select.stories-doc.mdx:
```diff
@@ -59,17 +52,11 @@ Dit kan je doen door de opties de spreaden in een nieuwe array (`[...options]`).
+Indien er minder dan 7 opties zijn raden we aan checkboxes of radio buttons te gebruiken.
 
 ## Varianten
 
```
Jammer genoeg door hoe onze storybook werkt, is het niet duidelijk hier voor de afnemer hoe deze varianten te implementeren. Behalve `Verwijderbaar` is uit het code voorbeelden niet te zien hoe je bv. `Groups` of `Read Only` gebruikt. Mogelijks `## Select opties` uitbreiden met model van `SelectOption` met annotaties?

Goldflow commented on libs/form/src/next/select/vl-select.component.ts:
```diff
@@ -46,347 +31,174 @@ export class VlSelectComponent extends FormControl {
-    };
+    private getGroupedOptions(): Record<string, SelectOption[]> {
+        const groupedOptions = this.options.reduce((groups, option) => {
+            const group = option.group || this.DEFAULT_GROUP_LABEL;
```
In deze opzet worden alle opties een `group` terwijl het volgens mij mogelijk is om `group`'s in combinatie met gewone `option`'s te gebruiken (hoewel ik daar niet direct een use-case voor kan vinden).

ksvlaanderen replied:
Dit heb ik zo gedaan zodat opties zonder groep altijd samen getoond worden.

Goldflow commented on libs/form/src/next/select/vl-select.component.ts:
```diff
@@ -46,347 +31,174 @@ export class VlSelectComponent extends FormControl {
-                .filter((option) => option.value)
-                .map((option) => option.value)
-        );
+        return Object.entries(groupedOptions).map(([group, options]) => {
```
In principe kan een `<optgroup>` ook `disabled` zijn. Als je dat niet aanbiedt kan de afnemer idd wel de individuele opties disablen maar niet de `group` zelf. Ik weet niet of dit nodig is.

Goldflow replied:
Zoals besproken, niet relevant voor single select want het ziet er altijd al `disabled` uit.

Goldflow reviewed (dismissed):
Good job! Ik denk dat naar documentatie dit alvast veel duidelijker is.

ksvlaanderen committed:
feat: UIG-2974 - native vl-select - introductie component

native vl-select-next component toegevoegd
uitgebreide vl-select hernoemd naar vl-select-rich
form-demo uitgebreid
Storybook en Cypress
